### PR TITLE
BREAKING CHANGE: supporting only node above 12

### LINKED ISSUE #83: Node v8: `Unexpected token {`
@ljharb I feel like you might know the answer for this.

I was looking at the #80 Looks like for node v12,16 it does nor matter it defaults to index.js and is able to use un-compiled code directly from `index.js`  

If I test with node v8 I get `Unexpected token {` regardless of whether I have `dist` folder included or not.
It looks like to me that the `build` & `prepublishOnly` scripts don't really do much

```
    "build": "rm -fr ./dist && babel ./lib --out-dir ./dist --copy-files",
    "prepublishOnly": "npm test && npm run build"
```

Now what is the right way to support v8 in this case? Thank you 🙏

### PR DESCRIPTION
**Pull request by DianaSuvorova**
It is time to move on... Node8 is not supported starting [eslint@7](https://eslint.org/docs/user-guide/migrating-to-7.0.0#-nodejs-8-is-no-longer-supported) 
closes #83

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -30,23 +28,24 @@
     "babel-plugin-transform-object-rest-spread": "^6.26.0",
     "babel-preset-env": "^1.7.0",
     "babel-register": "^6.26.0",
-    "eslint": "^4.14.0",
+    "eslint": "^4.19.1",
```
Comment (ljharb) [inline package.json:31]: This should be updated

File: .github/workflows/tests.yml [hunk 2]
```diff
@@ -13,7 +13,7 @@ jobs:
 
     strategy:
       matrix:
-        node-version: [8.x, 10.x, 12.x]
+        node-version: [10.x, 12.x]
```
Comment (ljharb) [inline .github/workflows/tests.yml:16]:
```suggestion
        node-version: [10.x, 12.x, 14.x, 16.x]
```

File: package.json (hunk 1 above)
Comment (DianaSuvorova) [reply package.json:31]: and I would 100% agree with you. Let me dot it

File: package.json [hunk 3]
```diff
@@ -9,51 +9,49 @@
...
     "eslint-plugin-standard": "^3.0.1",
-    "husky": "^0.14.3",
-    "mocha": "^4.0.1",
+    "husky": "^7.0.4",
+    "mocha": "^9.2.0",
     "semantic-release": "^17.2.3"
   },
   "peerDependencies": {
-    "eslint-plugin-react": "^7.16.0"
+    "eslint-plugin-react": "^7.16.0",
```
Comment (ljharb) [inline package.json:42]: might as well hoist this up to the latest version of eslint-plugin-react also, since you're doing a breaking change anyways

Comment (DianaSuvorova) [reply package.json:42]: you don't have days off do you 😆

Comment (ljharb) [reply package.json:42]: every hour's an hour off from lots of things :-p there's just more than 24 hours worth of things to do in a day

File: package.json [hunk 4]
```diff
@@ -39,7 +40,6 @@
     "semantic-release": "^17.2.3"
   },
   "peerDependencies": {
-    "eslint-plugin-react": "^7.16.0",
```
Comment (ljharb) [inline package.json]: this still does need to be a peer dep tho, otherwise your users might have a duplicate version of it

**Comment by DianaSuvorova** [comment]
@ljharb I think this should do it. lmk if you see anything off. Thank you 🙏

File: .github/workflows/tests.yml [hunk 5]
```diff
@@ -13,7 +13,7 @@ jobs:
 
     strategy:
       matrix:
-        node-version: [8.x, 10.x, 12.x]
+        node-version: [12.x, 14.x, 16.x]
```
Comment (ljharb) [inline .github/workflows/tests.yml:16]:
```suggestion
        node-version: [12.x, 14.x, 16.x, 17.x]
```

**Review by ljharb — APPROVED** [review]

**Pull request merged**